Profile Page.

pandilladivya opened the pull request:
**The following are the tasks finished in this PR**

-> User profile page.
-> Modified Manual Signup page to take first name and last name
-> Integrated 'users' node with firebase.
-> AutoFilled Data in Event Registration Page.

@sandarumk @HimanshiJain 
Kindly Review.

HimanshiJain commented:
So there is an inconsistency here. For events where volunteers are not required, I should not be able to register as a volunteer right? In fact I should not be able to select that radio button corresponding to "Volunteer"?

pandilladivya commented:
1. Yeah! I forgot to handle that case. I'll edit.
2. Do you want me to add Edit button in profile page. Since we never dicussed about it. I have made the basic one.

HimanshiJain commented:
So how are we able to add Affiliations? Can't the same be applied to name instead of an edit button?

pandilladivya commented:
We are not autofilling affiliations.
We are only doing it for email first name and last name.
Because we don't get affiliations when we signin with google.

We will have to modify user's node in firebase after modifying, So I think we need to have one edit button.

HimanshiJain commented:
So if I add my affiliation to the profile page, will it be saved in Firebase?

pandilladivya commented:
Currently,No. Thought we should discuss with @MeepyMay regarding requirements and fields to be included in profile page. ( we didn't get any requirement to add profile page )

Main Reasons to include profile page is that User should be able to Logout from the application.

HimanshiJain commented:
So, we will discuss with May but how do you think the mechanism should be? What do you think should be the most appropriate way to manage profile and editions (including which fields should be editable), etc.

pandilladivya commented:
I think, Except email everything can be editable. 
Once any of the other fields are edited, we can show a "SAVE" button. If the user clicks it, we can query firebase and update the changes.

HimanshiJain commented:
Cool! Let's do this.

MeepyMay commented:
- First Name
- Last Name
- Title (i.e. Program Manager, Software Engineer, Intern, Student, etc.)
- Affiliation/Organization (i.e. Microsoft, Anita Borg Institute, etc.) 

Also, yes as @pandilladivya said. All but email should be editable. :) Great job Divya.

pandilladivya commented:
The following is the demo of the work.

![ezgif com-video-to-gif 15](https://user-images.githubusercontent.com/16818129/29579783-732532ec-8791-11e7-8b87-4f1f003a1b91.gif)

pandilladivya commented:
@sandarumk @HimanshiJain 
Kindly Review

pandilladivya commented:
@HimanshiJain @sandarumk 
Disabled Radio button for events which do not require volunteers.
The following is the demo

![ezgif com-video-to-gif 16](https://user-images.githubusercontent.com/16818129/29581521-2f94364e-8797-11e7-89a1-9772ccaf6c50.gif)

HimanshiJain commented:
The demo looks great! I can see the volunteer radio button disabled but whenever I open the profile page, the app crashes on one device and on the other device the activity finishes and I get back to the same position. Please look into this.

pandilladivya commented:
@HimanshiJain I will look into it.

pandilladivya commented:
@sandarumk @HimanshiJain 
Kindly review.
I have fixed it.
:)

sandarumk commented on app/src/main/res/layout/content_profile.xml:
```diff
@@ -0,0 +1,162 @@
+                android:textSize="@dimen/font_headings" />
+
+            <EditText
+                android:id="@+id/input_fname"
```
The edit text should be single line for first name and last name.

sandarumk replied:
Similar to title as well

pandilladivya replied:
Okay

pandilladivya commented:
@sandarumk Updated!
@HimanshiJain @sandarumk Kindly review. 😊

sandarumk commented:
When I access the app without signing in, and go to the profile page, it will have the logout button only. @HimanshiJain Can you check this with your device?
Pressing "About VOLA" gives a "Logout Toast"
What happens if the user has no internet? For me when I try to register without internet it gives registration error toast. In that case can you give connect to internet or that kind of error?
Also when the credentials are wrong the spinner keeps spinning at verifying user.
What if the user is already registered and try to sign up again? Instead of using error messages as toasts, can you use a snack bar? https://material.io/guidelines/patterns/errors.html#errors-incompatible-state-errors @HimanshiJain What are your thoughts on this?

pandilladivya commented:
Yeah. I agree.
Snackbar would be a really good replacement for Toasts.
I think we have to give error message if the user is not connected to internet.
I will update it.

HimanshiJain commented:
1. I think profile page should only show Login button when the user is not logged in. (I think Dinu also meant that and by mistake typed Logout) or the other option could be remove this option from the Navigation Drawer when the user is not logged in. which do you think is better @sandarumk 
2. Right Remove that toast and  fix the spinner @pandilladivya .
3. For without internet thing, I think we should do that in a separate PR, because that needs to be done for the complete application from events list to profile or registration to sign in/sign up to everything because the application fetches all the data from Firebase. Actually since the all the features are dependent on Firebase and thus internet, should we check the internet as soon as the app is initiated? what would you suggest @sandarumk 
4. Snack bar will definitely look better for some cases.

sandarumk commented:
I think removing the access altogether to the profile page is good for non registered user.

pandilladivya commented:
@sandarumk @HimanshiJain 
Can we show an alert in profile page if the user is not loggedin?
Which displays the info message that the user needs to login.
And navigate the user to login page if he click ok. else navigate to home page.
What do you suggest?

HimanshiJain commented:
That can also be done. Our aim is that a user who hasn't logged in should not be able to see the profile page and definitely not the fields or logout button as are shown currently. Because that doesn't make sense.

pandilladivya commented:
I have also handled Landscape mode UI issues.
Added an alert in the profile page if the user is not logged in.
Removed "Navigate Me" option from the menu, since we are not using it.
Removed Toast on " About VOLA!" is clicked.

HimanshiJain commented on app/src/main/java/vola/systers/com/android/activities/SignInActivity.java:
```diff
@@ -177,38 +219,56 @@ private void launchHomeScreen() {
     private void googleSignIn() {
         Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
-        startActivityForResult(signInIntent, RC_GOOGLE_SIGN_IN);
+        if(! new NetworkConnectivity().checkConnectivity(this)) {
```
Instead of checking the network connectivity in each function, shouldn't we check it for any click that happens. That is before the switch case in the onClick function? So that if the internet is not connected, we do not enter the switch case and just show the snackbar?

pandilladivya replied:
Yeah, but there might be the case like I have internet when i enter the screen and after sometime i click signup button when I don't have connectivity.
To handle such cases I have used this.

HimanshiJain replied:
So when the user clicks any button, since we have managed all the clicks through a single function, even if the screen loads, if we check the connectivity in the onClick function, we will be able to check right?

pandilladivya replied:
YES! Sorry I misunderstood your question.. Yeah that is a good idea!
Will change 😊

pandilladivya replied:
But we should also check the switch case because navigating to the signup screen or navigating to the home screen should work even with out internet.

pandilladivya replied:
Done! Ammended my changes to previous commit

HimanshiJain commented on app/src/main/res/layout/content_event_detail_view.xml:
```diff
@@ -11,6 +11,13 @@
         android:layout_width="fill_parent"
         android:layout_height="wrap_content">
+
+    <android.support.design.widget.CoordinatorLayout
```
Indentation. in other layouts files too.

HimanshiJain commented on app/src/main/java/vola/systers/com/android/activities/SignInActivity.java:
```diff
@@ -52,8 +54,9 @@
 
     private static final int RC_GOOGLE_SIGN_IN = 007;
     private static final int RC_FACEBOOK_SIGN_IN=64206;
+    ProgressDialog progressDialog ;
```
Done this?

HimanshiJain commented:
@pandilladivya Is this last commit about saving events to calendar related to this PR? Shouldn't it be a different PR?

sandarumk commented:
Nope. Please add the commits related to the profile page and issues we asked. We are already late on this PR

pandilladivya commented:
Removed that commit!

sandarumk commented:
@pandilladivya what about the other fixes?

pandilladivya commented:
@sandarumk @HimanshiJain Done with all the changes! 😊
